Services/LDAP/classes/class.ilLDAPRoleAssignmentRule.php

```diff
@@ -125,10 +125,30 @@ public function matches(array $a_user_data): bool
     protected function wildcardCompare(string $a_str1, string $a_str2): bool
     {
         $pattern = str_replace('*', '.*?', $a_str1);
-        $this->logger->debug(': Replace pattern:' . $pattern . ' => ' . $a_str2);
-        return preg_match('/^' . $pattern . '$/i', $a_str2) === 1;
+
+        foreach (['/', '~', '@', ';', '%', '`', '#'] as $delimiter) {
+            $pattern_string = $delimiter . "^" . $pattern . '$' . $delimiter . 'i';
+            $this->logger->debug(': Match pattern:' . $pattern . ' => ' . $a_str2);
+
+            set_error_handler(static function (int $severity, string $message, string $file, int $line): never {
+                throw new ErrorException($message, $severity, $severity, $file, $line);
+            });
+
+            try {
+                if (preg_match($pattern_string, $a_str2) === 1) {
+                    return true;
+                }
```

Comment:
Please leave the expression as it was. This PR is just about catching errors and trying the next delimiter, if necessary.